selfhost/codegen.jakt

```diff
@@ -1507,6 +1507,13 @@ struct CodeGenerator {
 
             yield output
         }
+        Function(params, block) => {
+            mut generated_params: [String] = []
+            for param in params.iterator() {
+                generated_params.push(format("{} {}", .codegen_type(param.variable.type_id), param.variable.name))
+            }
+            yield format("[&]({}){}", join(generated_params, separator: ", "), .codegen_block(block))
```

Comment:
So right now we're just spitting out `[&]` in C++ codegen to capture *everything* by reference.

Let's force the user to be explicit about what's being captured, instead of doing it implicitly. Also, Jakt needs to be aware of captures, so that they can be used inside the lambda. Otherwise there's no point to capturing in the first place :)

Reply:
How should the syntax look? rip off cpp's syntax and pass it through to codegen or something else?